lms/views/helpers/frontend_app.py

```diff
@@ -12,42 +12,48 @@ def configure_grading(request, js_config):
     Note that this is entirely distinct from Canvas Speedgrader, which provides
     its own UI.
     """
+
     if (
-        request.lti_user.is_instructor
-        and _is_assignment_gradable(request)
-        and request.params.get("tool_consumer_info_product_family_code") != "canvas"
+        not request.lti_user.is_instructor
+        or not _is_assignment_gradable(request)
+        or request.params.get("tool_consumer_info_product_family_code") == "canvas"
     ):
-        js_config["lmsGrader"] = True
+        return
+
+    js_config["lmsGrader"] = True
 
     js_config["grading"] = {
         "courseName": request.params.get("context_title"),
         "assignmentName": request.params.get("resource_link_title"),
-    }
-
-    lis_result_sourcedid_svc = request.find_service(name="lis_result_sourcedid")
-    lis_result_sourcedids = lis_result_sourcedid_svc.fetch_students_by_assignment(
-        oauth_consumer_key=request.lti_user.oauth_consumer_key,
-        context_id=request.params.get("context_id"),
-        resource_link_id=request.params.get("resource_link_id"),
-    )
-    students = []
-    for student in lis_result_sourcedids:
-        # Using ``HUser`` NamedTuple to get at the ``userid`` prop
-        h_user = HUser(
-            authority=request.registry.settings["h_authority"],
-            username=student.h_username,
-            display_name=student.h_display_name,
-        )
-        students.append(
+        "students": [
             {
                 "userid": h_user.userid,
                 "displayName": h_user.display_name,
                 "LISResultSourcedId": student.lis_result_sourcedid,
                 "LISOutcomeServiceUrl": student.lis_outcome_service_url,
             }
-        )
+            for student, h_user in _find_students_for_assignment(
+                context_id=request.params.get("context_id"),
+                resource_link_id=request.params.get("resource_link_id"),
+                request=request,
+            )
+        ],
```

Comment:
:+1: I like the use of a list comprehension in a dict literal to simplify the code here